lib/fs.js

```diff
@@ -2070,11 +2074,43 @@ function writeAll(fd, isUserFd, buffer, offset, length, signal, callback) {
     } else {
       offset += written;
       length -= written;
-      writeAll(fd, isUserFd, buffer, offset, length, signal, callback);
+      writeAll(
+        fd, isUserFd, buffer, offset, length, signal, encoding, callback);
     }
   });
 }
 
+async function writeAllCustomIterable(
+  fd, isUserFd, buffer, offset, length, signal, encoding, callback) {
+  const result = await buffer.next();
+  if (result.done) {
+    fs.close(fd, callback);
+    return;
+  }
+  const resultValue = result.value.toString();
+  fs.write(fd, resultValue, undefined,
+           isArrayBufferView(buffer) ? resultValue.byteLength : encoding,
+           null, (writeErr, _) => {
+             if (writeErr) {
+               handleWriteAllErrorCallback(fd, isUserFd, writeErr, callback);
+             } else {
+               writeAll(fd, isUserFd, buffer, offset,
```

Comment:
Also, I think that it makes more sense to call `writeAllCustomIterable` instead of `writeAll` - though in that case, it would make sense to add an abort signal check to the start of the method.